src/rpc/core_rpc_server.cpp

```diff
@@ -1958,7 +1958,11 @@ namespace cryptonote
       m_was_bootstrap_ever_used = true;
     }
 
-    r = r && res.status == CORE_RPC_STATUS_OK;
+    if (r && res.status != CORE_RPC_STATUS_PAYMENT_REQUIRED && res.status != CORE_RPC_STATUS_OK)
```

Comment:
I don't understand the logic here, the call still failed due to lack of payment. So isn't the provided `struct` untouched?

Reply:
Yes, but since it returns failure, the caller would then try to run the RPC itself as a falllback, which is not what you want in this case.